packages/react/src/tanstackrouter.ts

```diff
@@ -0,0 +1,167 @@
+import { WINDOW, startBrowserTracingNavigationSpan, startBrowserTracingPageLoadSpan } from '@sentry/browser';
+import {
+  SEMANTIC_ATTRIBUTE_SENTRY_OP,
+  SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN,
+  SEMANTIC_ATTRIBUTE_SENTRY_SOURCE,
+} from '@sentry/core';
+
+import { browserTracingIntegration as originalBrowserTracingIntegration } from '@sentry/browser';
+import type { Integration } from '@sentry/types';
+
+// The following types are vendored types from TanStack Router, so we don't have to depend on the actual package
+
+export interface VendoredTanstackRouter {
```

Comment:
let's move this into a separate file and copy over license as well, I think we need to give attribution.